test/Test/Database/LSMTree/Internal/Run.hs

```diff
@@ -187,6 +190,34 @@ pagesContainEntries bsBlobs (page : pages) kops
   where
     (kopsHere, kopsRest) = splitAt (fromIntegral (rawPageNumKeys page)) kops
 
+-- | Runs in IO, but using a mock file system.
+--
+-- TODO: Also test file system errors.
+prop_WriteAndLoad :: WriteBuffer Key Val Blob -> Property
+prop_WriteAndLoad wb = ioProperty $ do
+    fs <- FsSim.mkSimErrorHasFS' FsSim.empty FsSim.emptyErrors
+    -- flush write buffer
+    let fsPaths = RunFsPaths 1337
+    written <- fromWriteBuffer fs fsPaths wb
+    loaded <- openFromDisk fs fsPaths
+
+    (1 @=?) =<< readIORef (lsmRunRefCount written)
+    (1 @=?) =<< readIORef (lsmRunRefCount loaded)
+
+    lsmRunNumEntries written @=? lsmRunNumEntries loaded
+    lsmRunFilter written @=? lsmRunFilter loaded
+    lsmRunIndex written @=? lsmRunIndex loaded
+
+    join $
+      assertEqual "k/ops file"
+        <$> FS.hGetAll fs (lsmRunKOpsFile written)
+        <*> FS.hGetAll fs (lsmRunKOpsFile loaded)
+    join $
+      assertEqual "blob file"
+        <$> FS.hGetAll fs (lsmRunBlobFile written)
+        <*> FS.hGetAll fs (lsmRunBlobFile loaded)
```

Comment:
Would it be sufficient to check whether `written` and `loaded` k/op and blob files point to the same path?

Reply:
Yes. :+1: